app.js

```diff
@@ -0,0 +1,47 @@
+const express = require('express');
+var flash = require('connect-flash');
+var session = require('express-session');
+var passport = require('passport');
+var flash = require('connect-flash');
+var cookieParser = require('cookie-parser');
+var morgan = require('morgan');
+const app = express();
+
+//Middleware
+//ejs
+app.set("view engine", "ejs");
+app.use(express.static(__dirname + '/public'));
+//Morgan
+app.use(morgan('dev'));
+//Cookie Parser
+app.use(cookieParser());
+require('./config/passport')(passport);
+
+//Body parser
+app.use(express.urlencoded({extended: false}));
+app.use(session({
+    secret: 'justasecret',
+    resave:false,
+    saveUninitialized: false
+}));
+   
+app.use(passport.initialize());
+app.use(passport.session());
+app.use(flash());
+
+//Global variables
+app.use(async function(req, res, next){
```

Comment:
I would export middleware into a separate function (or even a file), as example:
```const userMiddleware = async function (req, res, next) { /.../ }```
And use it as:
```app.use(userMiddleware);```